### DISCUSSION TITLE
Babel transpile JSLint for Rhino

### DISCUSSION THREAD

**Discussion by tntim96** [mozilla/rhino#1347] [Q&A] [answered]
I was interested to see if [JSLint](https://github.com/jslint-org/jslint/blob/beta/jslint.mjs) would run in Rhino.

It looks like a babel preset has been added in [Add support for rhino as a compilation target](https://github.com/babel/babel/pull/13448), however, I'm unclear how to test this out online. When I try https://babeljs.io/ with JSLint and add the target "rhino 1.7.14" the error message is:
```
[BABEL] /repl.jsx: Unknown browser rhino (While processing: "base$0")
```

**Comment by tntim96** [comment]
Saw this has also been discussed in #661 . Are you able to help @gausie or @gbrail ?

**Comment by p-bakker** [comment]
Specifying your target as an object should work (as per the [docs](https://babeljs.io/docs/options#targets)): 
```
"targets": {"rhino": "1.7.14"}
```

**Reply by tntim96** [reply]
If I try that I get

`Unknown browser rhino`

**Reply by p-bakker** [reply]
Which version of babel are you trying with?

**Comment by tntim96** [comment]
I've put some code in https://github.com/tntim96/jslint-transpile
Add to `.browserslistrc`
```
rhino 1.7.14
```
...and run
```
npm run compile
```

**Reply by p-bakker** [reply]
Think your babel.config.json should look like this:
```
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "rhino": "1.7.14"
      }
    ]
  ]
}
```

**Comment by tntim96** [comment]
Yep - tried that too. Same result

**Comment by tntim96** [comment, marked as answer]
This config seems to have worked
```
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "useBuiltIns": "entry",
        "corejs": "3.31",
        "targets": {
          "rhino": "1.7.14"
        },
        "modules": false
      }
    ]
  ]
}
```
I've pushed this to the repo too

**Reply by p-bakker** [reply]
k, so it works now? If so please access this as the answer
